### PR TITLE
More e2e testing and documentation

### PR DESCRIPTION
**Pull request by NicholasBottone**
- Made the README and the specs a lil fancier
- Added more e2e tests with Playwright:
  - Friends page
  - Play as guest
  - Navbar
- Added backend API test with Playwright:
  - Security (401 when unauthorized)

### CODE REVIEW

**Review by rkoch2 — APPROVED** [review]
Approving for merge!

File: README.md [hunk 1]
```diff
@@ -16,6 +19,21 @@ This is a [Next.js](https://nextjs.org/) project bootstrapped with [`create-next
 - To run the test suites: `npm test`
 - To run the linting rules: `npm run lint`
 
+## Environment Variables
```
Comment (andrew7li) [inline README.md:22]: I like the documentation!

File: tests/2fa.ts [hunk 2]
```diff
@@ -1,15 +1,54 @@
 import speakeasy from "speakeasy";
```
Comment (andrew7li) [inline tests/2fa.ts:1]: Maybe you could provide a description of what this test file is testing for?

File: tests/e2e/playAsGuest.spec.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
+import { test, expect } from "@playwright/test";
+
+test("Play puzzle as guest and lose", async ({ page }) => {
```
Comment (andrew7li) [inline tests/e2e/playAsGuest.spec.ts:3]: Is there a play as guest and win test?

**Review by andrew7li — APPROVED** [review]

File: tests/2fa.ts (hunk 2 above)
Comment (NicholasBottone) [reply tests/2fa.ts:1]: This is not a test file, this is a utility file

File: tests/e2e/playAsGuest.spec.ts (hunk 3 above)
Comment (NicholasBottone) [reply tests/e2e/playAsGuest.spec.ts:3]: There is not unfortunately since I do not know how to predict what the word will be for the guest puzzles

**Pull request merged**